libqtile/widget/base.py

```diff
@@ -342,10 +349,22 @@ def call_process(self, command, **kwargs):
 
     def _remove_dead_timers(self):
         """Remove completed and cancelled timers from the list."""
+        def is_ready(timer):
+            return timer in self.qtile._eventloop._ready
+
         self._futures = [
             timer
             for timer in self._futures
-            if not (timer.cancelled() or timer.when() < self.qtile._eventloop.time())
+            # Filter out certain handles...
+            if not (
+                timer.cancelled()
+                # Once a scheduled timer is ready to be run its _scheduled flag is set to False
+                # and it's added to the loop's `_ready` queue
+                or (isinstance(timer, asyncio.TimerHandle) and not timer._scheduled and not is_ready(timer))
+                # Callbacks scheduled via `call_soon` are put into the loop's `_ready` queue
+                # and are removed once they've been executed
+                or (isinstance(timer, asyncio.Handle) and not is_ready(timer))
+            )
```

Comment:
This could probably be re-written more neatly but I think this is more understandable like this. Happy to hear differing views.